### PR TITLE
Updating errors and information on the edge runtime

### PR DESCRIPTION
**Pull request by timeyoutakeit**
- Adds updated info about using GetServerSideProps and the Edge runtime
- Updates information on the switchable runtime
- Adds the runtime name to the table in the switchable runtime doc
- Updates the [Returning response body in Middleware](https://nextjs.org/docs/messages/returning-response-body-in-middleware) error doc to link to the fact that it is now possible in Next.js 13
- Fixes a broken link.

## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have a helpful link attached, see [`contributing.md`](https://github.com/vercel/next.js/blob/canary/contributing.md)

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] [e2e](https://github.com/vercel/next.js/blob/canary/contributing/core/testing.md#writing-tests-for-nextjs) tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see [`contributing.md`](https://github.com/vercel/next.js/blob/canary/contributing.md)

## Documentation / Examples

- [ ] Make sure the linting passes by running `pnpm build && pnpm lint`
- [ ] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: docs/basic-features/data-fetching/get-server-side-props.md [hunk 1]
```diff
@@ -45,6 +45,18 @@ It can be tempting to reach for an [API Route](/docs/api-routes/introduction.md)
...
+
+`getServerSideProps` can be used with both Serverless and Edge Runtimes, and you can set props in both. However, currently in Edge Runtime, you do not have access to the response object. This means that you cannot — for example — add cookies in `getServerSideProps`. To have access to the response object, you should continue to use the Serverless runtime, which is the default runtime.
+
+You can set the runtime on a [per-page basis](https://nextjs.org/docs/advanced-features/react-18/switchable-runtime#page-runtime-option) by modifying the `config`, for example:
+
+```
+export const config = {
+  runtime: 'nodejs'
+}
+```
```
Comment (feugy) [inline docs/basic-features/data-fetching/get-server-side-props.md:58]: **question:** given `nodejs` is the default value, should we document `experimental-edge` instead?

CC @ethomson

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 2]
```diff
@@ -38,22 +25,21 @@ export const config = {
...
+| Scalability                                                                                                                                         | /             | High              | Highest                                                  |
+| Security                                                                                                                                            | Normal        | High              | High                                                     |
+| Latency                                                                                                                                             | Normal        | Low               | Lowest                                                   |
+| Code Size                                                                                                                                           | /             | 50MB              | 1MB                                                      |
```
Comment (Kikobeats) [inline docs/advanced-features/react-18/switchable-runtime.md:39]: Code size for Edge Functions is now:

- hobby: 1MB
- oss/pro: 2MB
- enterprise: 4MB

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 3]
```diff
@@ -8,19 +8,6 @@ Next.js has two **server runtimes** where you can render parts of your applicati
 
 By default, Next.js uses the Node.js runtime. [Middleware](https://nextjs.org/docs/advanced-features/middleware) and [Edge API Routes](https://nextjs.org/docs/api-routes/edge-api-routes) use the Edge runtime.
 
-## Global Runtime Option
```
Comment (Kikobeats) [inline docs/advanced-features/react-18/switchable-runtime.md:11]: isn't this still valid? I mean, although it's out of experimental, a global configuration example could be convenient here.

File: docs/advanced-features/react-18/switchable-runtime.md (hunk 2 above)
Comment (Kikobeats) [inline docs/advanced-features/react-18/switchable-runtime.md:39]: Code size for Edge Functions is now:

- hobby: 1MB
- oss/pro: 2MB
- enterprise: 4MB

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 4]
```diff
@@ -8,19 +8,6 @@ Next.js has two **server runtimes** where you can render parts of your applicati
...
-
-To configure the runtime for your whole application, you can set the experimental option `runtime` in your `next.config.js` file:
-
-```js
-// next.config.js
-module.exports = {
```
Comment (Kikobeats) [inline docs/advanced-features/react-18/switchable-runtime.md:17]: if this has been moving out because isn't experimental anymore, could be still nice to keep this example to show how to setup global configuration 🙂

File: docs/advanced-features/react-18/switchable-runtime.md (hunk 2 above)
Comment (timeyoutakeit) [reply docs/advanced-features/react-18/switchable-runtime.md]: Chatted to @Kikobeats about this. The plan info isn't as relevant for folks using next.js, so we have stuck with the 1mb limit.

File: docs/basic-features/data-fetching/get-server-side-props.md [hunk 5]
```diff
@@ -45,6 +45,18 @@ It can be tempting to reach for an [API Route](/docs/api-routes/introduction.md)
 
 Take the following example. An API route is used to fetch some data from a CMS. That API route is then called directly from `getServerSideProps`. This produces an additional call, reducing performance. Instead, directly import the logic used inside your API Route into `getServerSideProps`. This could mean calling a CMS, database, or other API directly from inside `getServerSideProps`.
 
+### getServerSideProps with Edge API Routes
+
+`getServerSideProps` can be used with both Serverless and Edge Runtimes, and you can set props in both. However, currently in Edge Runtime, you do not have access to the response object. This means that you cannot — for example — add cookies in `getServerSideProps`. To have access to the response object, you should **continue to use the Serverless runtime**, which is the default runtime.
```
Comment (molebox) [inline docs/basic-features/data-fetching/get-server-side-props.md:50]: Is it serverless runtime? I thought its node runtime. Serverless is the name for execution model that lets you run server code without hosting a server

File: docs/basic-features/data-fetching/incremental-static-regeneration.md [hunk 6]
```diff
@@ -27,7 +27,7 @@ description: 'Learn how to create or update static pages at runtime with Increme
 
 Next.js allows you to create or update static pages _after_ you’ve built your site. Incremental Static Regeneration (ISR) enables you to use static-generation on a per-page basis, **without needing to rebuild the entire site**. With ISR, you can retain the benefits of static while scaling to millions of pages.
 
-> Note: the `experimental-edge` runtime (https://nextjs.org/docs/api-reference/edge-runtime) is currently not compatible with ISR although can leverage `stale-while-revalidate` by setting the `cache-control` header manually.
+> Note: the [`experimental-edge` runtime](https://nextjs.org/docs/api-reference/edge-runtime) is currently not compatible with ISR although can leverage `stale-while-revalidate` by setting the `cache-control` header manually.
```
Comment (molebox) [inline docs/basic-features/data-fetching/incremental-static-regeneration.md:30]:
```suggestion
> Note: The [`experimental-edge` runtime](https://nextjs.org/docs/api-reference/edge-runtime) is currently not compatible with ISR, although can leverage `stale-while-revalidate` by setting the `cache-control` header manually.
```

File: errors/returning-response-body-in-middleware.md [hunk 7]
```diff
@@ -1,5 +1,7 @@
 # Returning response body in Middleware
 
+> Note: In Next.js v13.0.0 you can now respond to middleware directly by returning a `NextResponse`. For more information, see [Producing a Response](https://nextjs.org/docs/advanced-features/middleware#producing-a-response).
```
Comment (molebox) [inline errors/returning-response-body-in-middleware.md:3]:
```suggestion
> Note: In Next.js v13.0.0 you can now respond to Middleware directly by returning a `NextResponse`. For more information, see [Producing a Response](https://nextjs.org/docs/advanced-features/middleware#producing-a-response).
```

File: docs/advanced-features/react-18/switchable-runtime.md [hunk 8]
```diff
@@ -8,19 +8,6 @@ Next.js has two **server runtimes** where you can render parts of your applicati
...
-  experimental: {
-    runtime: 'experimental-edge', // 'node.js' (default) | experimental-edge
-  },
-}
-```
-
 You can detect which runtime you're using by looking at the `process.env.NEXT_RUNTIME` Environment Variable during runtime, and examining the `options.nextRuntime` variable during compilation.
```
Comment (molebox) [inline docs/advanced-features/react-18/switchable-runtime.md:11]: Not part of this PR, but the cross usage of the word "runtime" confuses me. I know what this is saying, but i wonder if we could add a short callout that differentiates between the two meanings of the same word?

**Review by molebox — CHANGES_REQUESTED** [review]

File: docs/basic-features/data-fetching/get-server-side-props.md (hunk 5 above)
Comment (timeyoutakeit) [reply docs/basic-features/data-fetching/get-server-side-props.md:50]: Good spot!

**Comment by feugy** [comment]
Changes were cherry picked in #44045.
Thank you @timeyoutakeit!

**Comment by timneutkens** [comment]
Landed in https://github.com/vercel/next.js/pull/44045.
